Instant Search: update input box and icon styles

keoshi opened the pull request:
Related to issues reported in https://github.com/Automattic/jetpack/issues/14608; complementary PR to https://github.com/Automattic/jetpack/pull/14661

#### Changes proposed in this Pull Request:
* Update styling of main search box.
  * Ads `:hover` and `:focus` styles.
  * Using `px` units instead of `rem` to better control the output.
  * Clearer separation of input and icon.
  * Uses Gridicon/search icon.
* Revert search icon back to 24px.
* Fix bug with Gridicon's viewBox values reported in https://github.com/Automattic/jetpack/pull/14629#issuecomment-585474931.
* Remove magnifying glass options from Customizer as per https://github.com/Automattic/jetpack/issues/12540#issuecomment-586319510

![jps-search-box-v2](https://user-images.githubusercontent.com/390760/74660847-d4771000-518e-11ea-9daa-ea8a43104521.gif)


#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* No.

#### Testing instructions:
1. Follow [these setup instructions](https://github.com/Automattic/jetpack/blob/instant-search-master/modules/search/instant-search/README.md#testing-instructions) to set up Jetpack Instant Search on your site.
2. Search your site to trigger the Jetpack Search overlay.
3. Ensure that the Jetpack Search input field shows the icon and that both work correctly on all devices.

#### Proposed changelog entry for your changes:
* None.

keoshi committed:
Fix Gridicon's viewBox values

We shouldn't be messing with the values in viewBox at all since all icons have a baseline of 24 x 24 px. While sizing changes, their set up should be consistent.

Reported in: https://github.com/Automattic/jetpack/pull/14629#issuecomment-585474931

Thanks for catching this @jsnmoon !

keoshi committed:
Revert search icon back to 24px

keoshi committed:
Update syling of main search box

Complementary to https://github.com/Automattic/jetpack/pull/14661

gibrown reviewed (approved):
This looks so good.

jsnmoon commented on modules/search/instant-search/components/search-box.scss:
```diff
@@ -9,32 +10,41 @@ input.jetpack-instant-search__box-input.search-field,
 	background: #fff;
```
We might want to consider adding something to override this styling set by the 2020 theme:

```css
input[type="search"]:focus {
    outline: thin dotted;
    outline-offset: -4px;
}
```

`outline: none` applied to `input[type="search"].jetpack-instant-search__box-input` is a viable solution for this.

jsnmoon replied:
It's probably more appropriate to tackle this in a future PR referenced here: https://github.com/Automattic/jetpack/pull/14664#issuecomment-586216217.

jsnmoon reviewed (commented):
I don't see any `:focus` or `:hover` styling implied by the original PR description. Is this intentional?

keoshi commented:
@jsnmoon The intention was not to override but to be in a better place to inherit whatever `:focus` and `:hover` styles might be set on the theme. We can add our own styles though, which would not be a bad idea at all. I'd rather tackle that once this PR has landed though, if that's ok — seems it would be easier that way.

jeffgolenski reviewed (changes requested):
@keoshi I'm not sure if this is supposed to be handled in this PR, but the icon appears to be vertically off-center. The focus on the box looks good though.

jsnmoon reviewed (changes requested):
It looks like the background for the icon doesn't quite cover the input when zoomed at 110%:
<img width="854" alt="Screen Shot 2020-02-14 at 3 59 22 PM" src="https://user-images.githubusercontent.com/4044428/74575110-04ca6e80-4f43-11ea-8615-59986bb4bf51.png">

Same happens when zoomed at 125%:
<img width="754" alt="Screen Shot 2020-02-14 at 3 59 12 PM" src="https://user-images.githubusercontent.com/4044428/74575114-05630500-4f43-11ea-8045-c192aa532a6f.png">

keoshi committed:
Remove magnifying glass logo options from the Customizer

keoshi committed:
Remove magnifying glass options from live reload

keoshi committed:
Stop passing unused showLogo option

keoshi committed:
Add generic search icon, reorder icons alphabetically

keoshi committed:
Rework markup on the search box

- Adds Gridicon and input inside label.
- Moves screen reader text to its own span.
- Uses generic search icon.

keoshi committed:
Updates styling for the search box and its elements

keoshi committed:
[not verified] Remove magnifying glass from overlay options

keoshi committed:
Merge branch 'instant-search-master' into update/instant-search-input-box-styles

keoshi commented:
This has been reworked and could use another round of reviews! Updated the description above to reflect the latest changes.

gibrown reviewed (approved):
Looks good. I am seeing a narrower width for the search box now, but that may get fixed in #14661 so I think we should merge this.

folletto commented:
Design wise, solid improvement 👍

jeffgolenski commented:
@keoshi I did a quick screen recording. Everything looks and feels super smooth. My only ??? is the search button that appears in the input when I use keyboard navigation. When I hit `tab` it displays for a milisecond then disappears. I'm not sure it's needed for the functionality of instance search, but the disappearing act is probably something we want to address for a11y.

https://cloudup.com/cWiBND5DiUb

Commit:
Fix transition focus bug

jsnmoon reviewed (approved):
Looks good! 

I pushed up a commit that fixes a transition focus bug causing this issue: https://github.com/Automattic/jetpack/pull/14664#issuecomment-587598863.

keoshi commented:
@jsnmoon Thanks so much for fixing that!! I ran into that issue a bunch of times but failed to report it — thanks for doing so @jeffgolenski. Team work! 🙌
